### PR TITLE
PR: Prevent subprocess.Popen calls to create visible consoles on Windows.

### LINKED ISSUE #3529: Spyder 3 opens a command prompt window when running certain commands
I updated to Spyder 3, and now every time I click Run to launch my theano code, it will open nvcc.exe in a separate Windows command prompt window. Nothing else has changed. 

**What steps will reproduce the problem?**
1.  Run some code that uses nvcc.exe (or probably any other compiler) in Spyder 2
2.  Update to Spyder 3
3.  repeating step 1 (exactly the same code) will cause a black command prompt windows to appear (launch of compiler.exe)

**What is the expected output? What do you see instead?**
I expect to see no external applications opened in separate command prompt windows
## Versions and main components
- Spyder Version: 3.0.0
- Python Version: 2.7.11
- Operating system: Windows 10 Anniversary Edition
- Cuda 7.5
- Theano 0.9 dev2
## Dependencies

jedi >=0.8.1     :  0.9.0 (OK)
matplotlib >=1.0 :  1.5.1 (OK)
nbconvert >=4.0  :  4.1.0 (OK)
numpy >=1.7      :  1.11.1 (OK)
pandas >=0.13.1  :  0.17.0 (OK)
pep8 >=0.6       :  1.7.0 (OK)
pyflakes >=0.5.0 :  1.0.0 (OK)
pygments >=2.0   :  2.1.1 (OK)
pylint >=0.25    :  1.5.4 (OK)
qtconsole >=4.2.0:  4.2.1 (OK)
rope >=0.9.4     :  0.9.4 (OK)
sphinx >=0.6.6   :  1.3.5 (OK)
sympy >=0.7.3    :  0.7.6.1 (OK)

### LINKED ISSUE #4932: Windows command prompts pop up when executing pypandoc function calls in IPython console
## Description

**What steps will reproduce the problem?**

In IPython console execute the following commands:

1. import pypandoc
2. pypandoc.convert_text('hello', format='markdown', to='html')

Windows command prompts pop up and disappear again (stealing the focus every time, making it impossible to work while a script using pypandoc runs in spyder).

**What is the expected output? What do you see instead?**

pypandoc.convert_text() should execute without command prompts popping up.
 
**Please provide any additional information below**

These commands execute in an external python console and an external IPython console as expected without displaying additional windows.
Spyder was installed as part of Anaconda. pypandoc 1.4 was installed from conda-forge, yielding pandoc 1.19.2.1 in conda list.

## Version and main components

* Spyder Version: 3.2.0 
* Python Version: 3.6.1
* Qt Versions: 5.6.2, PyQt5 5.6 on Windows

## Dependencies
```
pyflakes >=0.6.0 :  1.5.0 (OK)
pycodestyle >=2.3:  2.3.1 (OK)
pygments >=2.0   :  2.2.0 (OK)
pandas >=0.13.1  :  0.20.1 (OK)
numpy >=1.7      :  1.12.1 (OK)
sphinx >=0.6.6   :  1.5.6 (OK)
rope >=0.9.4     :  0.9.4-1 (OK)
jedi >=0.9.0     :  0.10.2 (OK)
nbconvert >=4.0  :  5.1.1 (OK)
sympy >=0.7.3    :  1.0 (OK)
cython >=0.21    :  0.25.2 (OK)
qtconsole >=4.2.0:  4.3.0 (OK)
IPython >=4.0    :  5.3.0 (OK)
pylint >=0.25    :  1.6.4 (OK)

```

### PR DESCRIPTION
**Pull request by andfoy**
Fixes #3529

### CODE REVIEW

File: spyder/utils/site/sitecustomize.py [hunk 1]
```diff
@@ -196,6 +196,27 @@ def execfile(filename, namespace):
 
 
 #==============================================================================
+# Prevent subprocess.Popen calls to create visible console windows on Windows.
+# See issue #4932
+#==============================================================================
+
```
Comment (ccordoba12) [inline spyder/utils/site/sitecustomize.py]: Remove blank

File: spyder/utils/site/sitecustomize.py [hunk 2]
```diff
@@ -196,6 +196,27 @@ def execfile(filename, namespace):
...
+#==============================================================================
+
+import subprocess
+
+if os.name == 'nt':
+    creation_flag = 0x08000000  # CREATE_NO_WINDOW
+else:
+    creation_flag = 0  # Default value
+
+
```
Comment (ccordoba12) [inline spyder/utils/site/sitecustomize.py]: Remove blank

File: spyder/utils/site/sitecustomize.py [hunk 3]
```diff
@@ -196,6 +196,27 @@ def execfile(filename, namespace):
...
+class SubprocessPopen(subprocess.Popen):
+    def __init__(self, *args, creationflags=0, **kwargs):
+        super(SubprocessPopen, self).__init__(
+            *args, creationflags=creation_flag, **kwargs)
+
+
```
Comment (ccordoba12) [inline spyder/utils/site/sitecustomize.py]: Remove blank

**Comment by ccordoba12** [comment]
@andfoy, this is exactly what's needed. Did you test it locally?

**Comment by andfoy** [comment]
@ccordoba12 Yes, I tested it locally on my Windows VM, and it works as expected!

**Comment by ccordoba12** [comment]
@andfoy, let's apply this to Windows only. I don't want to touch Linux or macOS with this because they don't have this problem.

So please put everything under `if os.name == 'nt':`

**Comment by andfoy** [comment]
@ccordoba12 Actually the error was related to the named argument syntax.

**Comment by ccordoba12** [comment]
Ok, this looks really good and all our tests are passing. Thanks @andfoy!

**Pull request merged**